src/pages/instances/Events.tsx

```diff
@@ -7,19 +7,48 @@ import { useQueryClient } from "@tanstack/react-query";
 import { queryKeys } from "util/queryKeys";
 import { useOperations } from "context/operationsProvider";
 import { useNotify } from "@canonical/react-components";
+import useEventListener from "util/useEventListener";
 
 const EVENT_HANDLER_DELAY = 250;
 const WS_RETRY_DELAY_MULTIPLIER = 250;
 const MAX_WS_CONNECTION_RETRIES = 5;
+const EVENT_WS_STALE_TIMEOUT_MS = 3_600_000;
 
 const Events: FC = () => {
   const { isAuthenticated } = useAuth();
   const eventQueue = useEventQueue();
   const queryClient = useQueryClient();
   const notify = useNotify();
   const [eventWs, setEventWs] = useState<WebSocket | null>(null);
+  const [reconnectWsId, setReconnectWsId] = useState(0);
+  const [wsOpenTime, setWsOpenTime] = useState(0);
   const { refetchOperations } = useOperations();
 
+  const getCurrentTime = () => {
+    return new Date().getTime();
+  };
+
+  const reconnectWsOnFocusTab = () => {
+    if (document.visibilityState !== "visible") {
+      return;
+    }
+    // Force reconnect if the ws is not open.
+    if (!eventWs) {
+      setReconnectWsId((prevId) => prevId + 1);
+      return;
+    }
```

Comment:
This should be enough still - no need for any additional state. Just check whether it's `null` on `visibilitychange` and call the `connectEventWs` function in that case:
```suggestion
    if (!eventWs) {
      connectEventWs();
      return;
    }
```

Reply:
This goes back to the problem that someone needs to close the socket on unmount, only a use effect can do it :D